src/Domain/Creators/BinaryCompactObject.cpp

```diff
@@ -153,6 +154,15 @@ BinaryCompactObject::BinaryCompactObject(
         "of Layer 1 enveloping Object B requires excising the interior of "
         "Object B");
   }
+  using domain::BoundaryConditions::is_none;
+  if (outer_boundary_condition_ != nullptr and
+      ((not excise_interior_A_ and not excise_interior_B_) and
+       (inner_boundary_condition_ == nullptr or
+        not is_none(inner_boundary_condition_)))) {
+    PARSE_ERROR(
+        context,
+        "Inner boundary condition must be None if not excising both A and B");
+  }
   if ((inner_boundary_condition_ != nullptr and
```

Comment:
Option to write this as "(outer == nullptr) != (inner == nullptr)" instead... totally up to you I'm happy either way

Reply:
I find that much harder to understand, honestly :)

Reply:
Turns out there's a thing for this: `xor`. I've pushed a separate fixup so it's easily dropped if people don't like it

Reply:
Oh yeah 😅